### ISSUE TITLE
[BUG] Concerning warning in latest bugfix compile for SKR v1.3 board

### ISSUE THREAD

**Issue by bill-orange** [MarlinFirmware/Marlin#21025]
### Bug Description

This is only a warning.  I felt that I should report it because it may reveal a deeper issue, possibly caused by an update to Platformio.  The following warning occurs on the latest Bugfix when compiled with the included config files.

```
Marlin\src\HAL\shared\Delay.cpp:54:23: warning: ISO C++17 does not allow 'register' storage class specifier [-Wregister]
   54 |     register uint32_t start = HW_REG(_DWT_CYCCNT) - ASM_CYCLES_PER_ITERATION, elapsed;
      |                       ^~~~~
Marlin\src\HAL\shared\Delay.cpp:54:79: warning: ISO C++17 does not allow 'register' storage class specifier [-Wregister]
   54 |     register uint32_t start = HW_REG(_DWT_CYCCNT) - ASM_CYCLES_PER_ITERATION, elapsed;
      |                                                                               ^~~~~~~
```

### Configuration Files

[Marlin.zip](https://github.com/MarlinFirmware/Marlin/files/5948072/Marlin.zip)

### Steps to Reproduce

1. Compile
2. Observe warning
3. Note, compile is otherwise successful

**Expected behavior:**

No scary warning

**Actual behavior:**

Concerning warning

#### Additional Information

The reference to `ISO C++17 does not allow 'register' storage class specifier` suggests to me that this may be the result of a change to the compiler that will cause a failure to compile when C++17 is adopted.

**Comment by thisiskeithb** [comment]
@sjasonsmith: What do you think? I tend to ignore warnings since none of my personal builds will build cleanly.

**Comment by bugfixin** [comment]
The register keyword is explicitly meaningless according to http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2015/n4340 so I think it should be (and is safe to be) removed here

**Comment by sjasonsmith** [comment]
@thisiskeithb I prefer to enable all warnings and treat them as errors. Marlin’s not there, but that’s what I expect of code I maintain professionally.

**Comment by sjasonsmith** [comment]
Most Marlin warnings come from external libraries. We should at least try to have the Marlin code itself warning free.

**Comment by ellensp** [comment]
FYI MEATPACK also uses register and produces warning on LPC176x
That is all internal to Marlin

**Comment by bill-orange** [comment]
I am also getting a new warning on a Melzi build involving how variables are declared.  I will post it as a separate issue later.

Edit #21029

**Comment by rhapsodyv** [comment]
I think this was just fixed with #21178. Bugfix should be ok now.

**Comment by nanotuxi** [comment]
> I think this was just fixed with #21178. Bugfix should be ok now.

Can confirm that.

**Comment by thisiskeithb** [comment]
Fix has been merged. Closing.
